### PR TITLE
Add a Request.cookies function

### PR DESCRIPTION
**Pull request by ScalaFanatic**
As discussed in #2943, it works with both HTTP/1 and HTTP/2.

### CODE REVIEW

**Comment by ScalaFanatic** [comment]
The scalafix error is
```
[error] [launcher] could not retrieve Scala 2.12.10: missing scala.Option
```
Should I just close and re-open the pull request to trigger another Travis build? I assume this is another one of those flaky issues with Travis.

File: tests/src/test/scala/org/http4s/MessageSpec.scala [hunk 1]
```diff
@@ -126,6 +126,40 @@ class MessageSpec extends Http4sSpec {
...
+        RequestCookie("test2", "value2"),
+        RequestCookie("test3", "value3"))
+
+      "be empty if there are no Cookie headers present" in {
+        Request(Method.GET).cookies mustEqual List.empty
+      }
+
+      "parse discrete HTTP/1 Cookie header(s) into corresponding RequestCookies" in {
+        val cookies = Header("Cookie", "test1=value1; test2=value2; test3=value3")
+        val request = Request(Method.GET, headers = Headers.of(cookies))
```
Comment (kevinmeredith) [inline tests/src/test/scala/org/http4s/MessageSpec.scala]: Does your test require specifying the HTTP version when constructing the [Request](https://github.com/http4s/http4s/blob/master/core/src/main/scala/org/http4s/Message.scala#L242)?

File: tests/src/test/scala/org/http4s/MessageSpec.scala [hunk 2]
```diff
@@ -126,6 +126,40 @@ class MessageSpec extends Http4sSpec {
...
+      "parse discrete HTTP/2 Cookie header(s) into corresponding RequestCookies" in {
+        val cookies = Headers.of(
+          Header("Cookie", "test1=value1"),
+          Header("Cookie", "test2=value2"),
+          Header("Cookie", "test3=value3"))
+        val request = Request(Method.GET, headers = cookies)
+        request.cookies mustEqual cookieList
+      }
+
+      "parse HTTP/1 and HTTP/2 Cookie headers on a single request into corresponding RequestCookies" in {
```
Comment (kevinmeredith) [inline tests/src/test/scala/org/http4s/MessageSpec.scala]: Same as https://github.com/http4s/http4s/pull/2965/files#r344472600

File: tests/src/test/scala/org/http4s/MessageSpec.scala (hunk 1 above)
Comment (ScalaFanatic) [reply tests/src/test/scala/org/http4s/MessageSpec.scala]: No. That should not impact the way the headers are handled. If I need to add that for semantics, I can, but this specifically refers to the way the cookies are being sent.

Comment (kevinmeredith) [reply tests/src/test/scala/org/http4s/MessageSpec.scala]: 👍

**Comment by rossabaker** [comment]
This is backported into #2992.

**Pull request closed without merging**